torch/utils/data/dataset.py

```diff
@@ -260,6 +265,116 @@ def __len__(self):
         return len(self.indices)
 
 
+class ChunkDataset(IterableDataset):
+    r"""Dataset which uses hierarchical sampling for efficient data reading.
+
+
+    ``ChunkDataset`` is a stateful dataset that supports hierarchical sampling and
+    efficient reading through chunks. A chunk is selected based on a sampling strategy
+    (first level) and its content is shuffled (second level) before producing
+    a batch or an example.
+
+    In this context, chunks could be files, contents of folder,
```

Comment:
contents of a folder

Reply:
locally fixed. will push after all feedbacks